llvm/lib/IR/Constants.cpp

```diff
@@ -2015,6 +2018,118 @@ Value *NoCFIValue::handleOperandChangeImpl(Value *From, Value *To) {
   return nullptr;
 }
 
+//---- ConstantPtrAuth::get() implementations.
+//
+
+static bool areEquivalentAddrDiscriminators(const Value *V1, const Value *V2,
+                                            const DataLayout &DL) {
+  APInt V1Off(DL.getPointerSizeInBits(), 0);
```

Comment:
Should be IndexSize, not PointerSize.

Reply:
Went with type size to allow the (somewhat unfortunate) cases of blend intrinsics taking i64, but the constant having pointers.  I could probably rejigger the whole body to have them consistently on a single index type